homeassistant/components/updater.py

```diff
@@ -0,0 +1,143 @@
+"""
+homeassistant.components.updater
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Indicates when a new update for HA is available
+"""
+# system imports
+import logging
+import os
+import requests
+import subprocess
+
+# homeassistant imports
+import homeassistant
+from homeassistant.helpers.entity import Entity
+
+# homeassistant constants
+DOMAIN = "updater"
+DEPENDENCIES = []
+REQUIREMENTS = ['gitpython>=1.0.1']
+CONF_PID_FILE = "pid_file"
+CONF_LOG_FILE = "log_file"
+CONF_REPO_NAME = "repo_name"
+CONF_REPO_BRANCH = "repo_branch"
+HA_SOURCE_DIR = os.path.abspath(
+    os.path.join(os.path.dirname(homeassistant.__file__), '..'))
+GH_API_CALL = 'https://api.github.com/repos/{repo}/commits?sha={branch}'
+
+# setup logger
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """
+    Setup updater component.
+    """
+    # get optional configuration
+    pid_file = config['updater'].get(CONF_PID_FILE, None)
+    log_file = config['updater'].get(CONF_LOG_FILE, None)
+    repo_name = config['updater'].get(CONF_REPO_NAME, 'balloob/home-assistant')
+    repo_branch = config['updater'].get(CONF_REPO_BRANCH, 'master')
+
+    # create updater service
+    def run_update(service_call):
+        ''' run the updater '''
+        upscript = os.path.join(HA_SOURCE_DIR, 'scripts', 'update')
+        mypid = str(os.getpid())
+        cmd = [upscript, HA_SOURCE_DIR, mypid]
+
+        if log_file is not None:
+            cmd.append(log_file)
+            if pid_file is not None:
+                cmd.append(pid_file)
+
+        subprocess.call(cmd)
+    hass.services.register(DOMAIN, 'update', run_update)
+
+    # find local copy info
+    try:
+        from git import Repo
+    except ImportError:
+        _LOGGER.error('Missing package gitpython')
+        current_sha = None
+    else:
+        repo = Repo(HA_SOURCE_DIR)
+        if repo.bare:
+            current_sha = None
+        else:
+            current_sha = repo.head.commit.hexsha
+
+    # create component entity
+    Updater(hass, current_sha, repo_name, repo_branch, pid_file, log_file)
+
+    return True
+
+
+class Updater(Entity):
+    """ Updater entity class """
+
+    hass = None
+    entity_id = 'updater.updater'
+    name = 'Updater'
+
+    def __init__(self, hass, current_sha, repo_name, branch, pid_file,
+                 log_file):
+        # pylint: disable=too-many-arguments
+        self.hass = hass
+        self.config = {'repo_name': repo_name, 'branch': branch,
+                       'pid_file': pid_file, 'log_file': log_file}
+
+        # register update check event
+        def check_ha_update(event):
+            """ Check GitHub for avaialable update """
+            self.update()
+        hass.track_time_change(check_ha_update, hour=0, minute=0, second=0)
+
+        # initialize data
+        self.versions = {'newest_sha': None, 'newest_msg': None,
```

Comment:
Specifically because pylint was yelling at me for having too many properties. It is kind of 6 one way, half dozen the other. I figured I would err on the side of not using another pylint exception.